language-examples/php/rest/getJwtToken.php

```diff
@@ -10,18 +10,24 @@
  */
 function get_token($auth_url, $client_id, $client_secret)
 {
-    $url = $auth_url . '/oauth2/token';
-    $encoded = base64_encode($client_id . ':' . $client_secret);
+    //remove trialing / from auth_url
+    $url = rtrim($auth_url, '/');
+
+    // append /oauth2/token to the auth_url if it does not end with it 
+    if (!str_ends_with($url, '/oauth2/token')) {
+        $auth_url .= '/oauth2/token';
+    }
 
     $fields = [
         'grant_type' => 'client_credentials',
         'client_id' => $client_id,
+        'client_secret' => $client_secret
     ];
 
     $ch = curl_init();
     curl_setopt($ch, CURLOPT_URL, $url);
     curl_setopt($ch, CURLOPT_POST, true);
-    curl_setopt($ch, CURLOPT_HTTPHEADER, ['Authorization: Basic ' . $encoded]);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/x-www-form-urlencoded']);
```

Comment:
I am curious. Why change to this type of auth? Was the bearer token auth not working?

Reply:
I just synced the header with other examples in NodeJs, hence adding 'Content-Type'. But it was not working with Basic Auth.